op-batcher/batcher/espresso.go

```diff
@@ -21,43 +22,529 @@ import (
+
+		// Submit the job to the worker
+		select {
+		case <-s.ctx.Done():
+			return
+
+		case worker <- espressoTransactionJobAttempt{job: job, resp: s.submitRespQueue}:
+		}
+	}
+}
+
+// scheduleVerifyReceiptJobs is a function that is meant to be run in a
+// goroutine.
+//
+// It handles the scheduling of verify receipt jobs so that the verify receipt
+// workers can process them.
+func (s *espressoTransactionSubmitter) scheduleVerifyReceiptsJobs() {
+	for {
+		var ok bool
+
+		// Get a worker from the worker queue
+		var worker chan espressoVerifyReceiptJobAttempt
+		select {
+		case <-s.ctx.Done():
+			return
+
+		case worker, ok = <-s.verifyReceiptWorkerQueue:
+			if !ok {
+				// Our channel is closed, and we are done
+				return
+			}
+		}
+
+		// Get a job from the job queue
+		var job espressoVerifyReceiptJob
+		select {
+		case <-s.ctx.Done():
+			return
+		case job, ok = <-s.verifyReceiptJobQueue:
+			if !ok {
+				// Our channel is closed, and we are done
+				return
+			}
+		}
+
+		// Submit the job to the worker
+		select {
+		case <-s.ctx.Done():
+			return
+
+		case worker <- espressoVerifyReceiptJobAttempt{job: job, resp: s.verifyReceiptRespQueue}:
+		}
+	}
+}
+
+// espressoSubmitTransactionWorker is a function that is meant to be run as a
+// goroutine.  It will create a channel for it's job queue, and submit those to
+// the worker queue in order to wait for work.  It will then take that job and
+// attempt to submit the transaction contained within to espresso using the
+// given espresso client. It will submit the response back to the channel
+// contained within the job attempt it received.
+//
+// It's lifetime is governed by the context passed to it, and it will stop
+// processing when that context is cancelled.
+//
+// NOTE: If the context is cancelled after a job has been received, but before
+// it is able to submit the transaction, or report about it's result, the job
+// may be lost.
+func espressoSubmitTransactionWorker(
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	cli espressoClient.EspressoClient,
+	workerQueue chan<- chan espressoTransactionJobAttempt,
+) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	defer wg.Done()
+	ch := make(chan espressoTransactionJobAttempt)
+	defer close(ch)
+
+	for {
+		var ok bool
+		select {
+		case <-ctx.Done():
+			return
+
+			// Queue our job queue, asking for work
+		case workerQueue <- ch:
+		}
+
+		// Wait for a job to run
+		var jobAttempt espressoTransactionJobAttempt
+		select {
+		case <-ctx.Done():
+			return
+		case jobAttempt, ok = <-ch:
+			if !ok {
+				// Our channel is closed, and we are done
+				return
+			}
+		}
+
+		// Submit the transaction to Espresso
+		hash, err := cli.SubmitTransaction(ctx, *jobAttempt.job.transaction)
+
+		jobAttempt.job.attempts++
+		resp := espressoSubmitTransactionJobResponse{
+			job:  jobAttempt.job,
+			hash: hash,
+			err:  err,
+		}
+
+		if jobAttempt.job.attempts > 3 {
+			// We have submitted this job quite a bit.
```

Comment:
Is this part of some TODO?

Reply:
Initially intended to be so, yes.  But perhaps this is best removed for now.

Reply:
Addressed in db23892adc18ad7fe5ff7c51cfffb1203b8cb3ac.